### PR TITLE
[AiLab] levelbuilders can include a pre-trained model in App Lab levels 

### PR DESCRIPTION
**Pull request by Erin007**
Levelbuilders will now have an additional setting on App Lab level edit pages to add a model name and model id. 
Screen Shot 2021-03-17 at 10 47 14 AM

If model name and model id are set for an App Lab level, that model will show up first in the Model Manager Dialog. 
Screen Shot 2021-03-17 at 3 00 49 PM

### CODE REVIEW

**Comment by Erin007** [comment]
NOTE: Model metadata is environment specific. It's stored in the UserMLModels database table. This means a **levelbuilder will need to train a model on production**, then copy the id and name to use when editing a level on levelbuilder.  I think this is ok, because it's kind of like how example projects work, but please let know if you have ideas for improvements.

File: apps/src/code-studio/components/ModelManagerDialog.jsx [hunk 1]
```diff
@@ -49,6 +51,9 @@ export default class ModelManagerDialog extends React.Component {
       url: '/api/v1/ml_models/names',
       method: 'GET'
     }).then(models => {
+      if (this.props.levelbuilderModel && this.props.levelbuilderModel.id) {
```
Comment (maddiedierker) [inline apps/src/code-studio/components/ModelManagerDialog.jsx]: this is a great place to use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)! (charlie just enabled this in #39543, so we couldn't use that feature until now. you'll likely need to pull/merge staging into your branch as well)

File: apps/src/lib/ui/SettingsCog.jsx [hunk 2]
```diff
@@ -135,6 +135,16 @@ class SettingsCog extends Component {
     return pageConstants && pageConstants.aiEnabled;
   }
 
+  levelbuilderModel() {
+    let model = {};
+    let pageConstants = getStore().getState().pageConstants;
+    if (pageConstants && pageConstants.aiModelId && pageConstants.aiModelName) {
```
Comment (maddiedierker) [inline apps/src/lib/ui/SettingsCog.jsx]: another good use-case for optional chaining! this conditional would become: `pageConstants?.aiModelId && pageConstants?.aiModelName`

File: apps/src/code-studio/components/ModelManagerDialog.jsx (hunk 1 above)
Comment (Erin007) [reply apps/src/code-studio/components/ModelManagerDialog.jsx]: Well, isn't that handy? Thanks Maddie (and Charlie!). Updating...

**Review by maddiedierker — APPROVED** [review]
🎉

**Pull request merged**